docs/tutorial.md

```diff
@@ -0,0 +1,372 @@
+
+Below code will initize and defines the informers/watches for your operator.
+
+```
+  @Override
+  public int run(String... args) throws Exception {
+    operator.start();
+
+    Quarkus.waitForExit();
+    return 0;
+  }
+```
+
+## Create a new API and Controller
+
+Create a new Custom Resource Definition (CRD) API with group `cache` version `v1alpha1` and Kind Memcached. Below messages are informational mesages.
+
+```console
+$ operator-sdk create api --plugins quarkus --group cache --version v1 --kind Memcached
+InjectResource called
+UpdateResource called
+Scaffold called
+NewCreateAPIScaffolder called
+InjectFS called
+api.Scaffold()
+PostScaffold called
+...
+```
+
+This will scaffold the MemcachedController, MemcachedSpec, MemcachedStatus and Memcached.
+
+#### Understanding Kubernetes APIs
+
+For an in-depth explanation of Kubernetes APIs and the group-version-kind model, check out these [kubebuilder docs][kb-doc-gkvs].
+
+The API's used from [java-operator-sdk][https://github.com/java-operator-sdk/java-operator-sdk]. It is similar to `Controller-runtime` from Go lang.
+
+### Define the API
+
+To begin, we will represent our API by defining the `Memcached` type, which will have a `MemcachedSpec.Size` field to set the quantity of memcached instances (CRs) to be deployed, and a `MemcachedStatus.Nodes` field to store a CR's Pod names.
+
+**Note** The Node field is just to illustrate an example of a Status field. In real cases, it would be recommended to use [Conditions][conditionals].
+
+
+```
+// MemcachedSpec defines the desired state of Memcached
+public class MemcachedSpec {
+
+    // Add Spec information here
+	// Size is the size of the memcached deployment
+    private Integer size;
+
+    public Integer getSize() {
+        return size;
+    }
+
+    public void setSize(Integer size) {
+        this.size = size;
+    }
+}
+```
+
+```
+import java.util.ArrayList;
+import java.util.List;
+
+// MemcachedStatus defines the observed state of Memcached
+public class MemcachedStatus {
+
+    // Add Status information here
+	// Nodes are the names of the memcached pods
+    private List<String> nodes;
+
+    public List<String> getNodes() {
+        if (nodes == null) {
+            nodes = new ArrayList<>();
+        }
+        return nodes;
+    }
+
+    public void setNodes(List<String> nodes) {
+        this.nodes = nodes;
+    }
+}
+```
+
+```
+import io.fabric8.kubernetes.api.model.Namespaced;
+import io.fabric8.kubernetes.client.CustomResource;
+import io.fabric8.kubernetes.model.annotation.Group;
+import io.fabric8.kubernetes.model.annotation.Version;
+
+// Memcached is the Schema for the memcacheds API
+
+@Version("v1alpha1")
+@Group("cache.example.com")
+public class Memcached extends CustomResource<MemcachedSpec, MemcachedStatus>
+    implements Namespaced {}
+```
```

Comment:
probably should explain a little what this is as well.

Reply:
Modified